rest/handler.go

```diff
@@ -203,11 +203,24 @@ func newHandler(server *ServerContext, privs handlerPrivs, serverType serverType
 // ctx returns the request-scoped context for logging/cancellation.
 func (h *handler) ctx() context.Context {
 	if h.rqCtx == nil {
-		h.rqCtx = base.CorrelationIDLogCtx(h.rq.Context(), h.formatSerialNumber())
+		ctx := base.CorrelationIDLogCtx(h.rq.Context(), h.formatSerialNumber())
```

Comment:
Reconstructing contexts is cheap but not free since it is does a copy. Should we change `RequestLogCtx` to set a correlation id also, so we only have to construct this once?